flow/layers/color_filter_layer_unittests.cc

```diff
@@ -94,17 +100,23 @@ TEST_F(ColorFilterLayerTest, SimpleFilter) {
   EXPECT_TRUE(layer->needs_painting(paint_context()));
   EXPECT_EQ(mock_layer->parent_matrix(), initial_transform);
 
-  SkPaint filter_paint;
-  filter_paint.setColorFilter(layer_filter);
-  layer->Paint(paint_context());
-  EXPECT_EQ(
-      mock_canvas().draw_calls(),
-      std::vector({MockCanvas::DrawCall{
-                       0, MockCanvas::SaveLayerData{child_bounds, filter_paint,
-                                                    nullptr, 1}},
-                   MockCanvas::DrawCall{
-                       1, MockCanvas::DrawPathData{child_path, child_paint}},
-                   MockCanvas::DrawCall{1, MockCanvas::RestoreData{0}}}));
+  DisplayListBuilder expected_builder;
+  /* ColorFilterLayer::Paint() */ {
+    expected_builder.setColorFilter(dl_color_filter.get());
+    expected_builder.saveLayer(&child_bounds, true);
+    {
+      /* MockLayer::Paint() */ {
+        expected_builder.setColor(SkColors::kYellow.toSkColor());
+        expected_builder.setColorFilter(nullptr);
+        expected_builder.drawPath(child_path);
```

Comment:
This would look simpler if you use the `drawPath(..., &dl_paint)` variant. You wouldn't have to manually reset the attributes in the builder (assuming you used different paint objects in each call). The same thing could be said for the saveLayer, though it wouldn't be any fewer lines of code. I'm steering the builder interface and its internal uses towards the call variants that take a DlPaint and may eventually make the stateful calls a hidden implementation detail not visible from the Builder API.

Note that the set methods on DlPaint are chainable for easy initialization and more compact code, as in:
```
DlPaint paint = DlPaint().setColor(DlColor.kYellow());
```